### PR TITLE
Rewrite Workspace errors as DiagnosticData structs

### PR DESCRIPTION
**Pull request by hartbit**
This change is required for future improvements to the Workspace support of the DiagnosticsEngine.

### CODE REVIEW

**Comment by hartbit** [comment]
@swift-ci please smoke test

File: Sources/Workspace/Diagnostics.swift [hunk 1]
```diff
@@ -42,3 +42,158 @@ extension ManifestParseError: DiagnosticDataConvertible {
         }
     }
 }
+
+extension Workspace {
```
Comment (aciidb0mb3r) [inline Sources/Workspace/Diagnostics.swift]: How about defining the diagnostics under an enum called `WorskpaceDiagnostics`, that way we don't need add the suffix "Diagnostic" in each struct. e.g.: `WorskpaceDiagnostics.UncommitedChanges`

**Comment by hartbit** [comment]
Moved to `WorkspaceDiagnostics`.

**Comment by hartbit** [comment]
@swift-ci please smoke test

**Comment by aciidb0mb3r** [comment]
@shahmishal can you check why @hartbit is unable to run CI?

**Comment by aciidb0mb3r** [comment]
@swift-ci please smoke test

**Comment by aciidb0mb3r** [comment]
@swift-ci please smoke test

**Comment by shahmishal** [comment]
> @shahmishal can you check why @hartbit is unable to run CI?
Now you should be able to.

**Comment by aciidb0mb3r** [comment]
Failure looks related

**Comment by aciidb0mb3r** [comment]
Thanks @shahmishal!

**Comment by hartbit** [comment]
@swift-ci please smoke test

**Comment by hartbit** [comment]
@swift-ci please smoke test

**Comment by hartbit** [comment]
@swift-ci please smoke test

**Comment by hartbit** [comment]
@swift-ci please smoke test OS X

**Comment by hartbit** [comment]
@swift-ci please smoke test

**Comment by hartbit** [comment]
@swift-ci please smoke test

**Pull request merged**